scipy/sparse/linalg/propack.py

```diff
@@ -0,0 +1,336 @@
+    compute_v : bool, optional
+        If `True` (default) then compute right singular vectors, `v`.
+    tol : float, optional
+        The desired relative accuracy for computed singular values.
+        If not specified, it will be set based on machine precision.
+    v0 : array_like, optional
+        Starting vector for iterations: must be of length ``A.shape[0]``.
+        If not specified, PROPACK will generate a starting vector.
+    full_output : bool, optional
+        If `True`, then return sigma_bound.  Default is `False`.
+    delta : float, optional
+        Level of orthogonality to maintain between Lanczos vectors.
+        Default is set based on machine precision.
+    eta : float, optional
+        Orthogonality cutoff.  During reorthogonalization, vectors with
+        component larger than `eta` along the Lanczos vector will be purged.
+        Default is set based on machine precision.
+    anorm : float, optional
+        Estimate of ``||A||``.  Default is `0`.
+    cgs : bool, optional
+        If `True`, reorthogonalization is done using classical Gram-Schmidt.
+        If `False` (default), it is done using modified Gram-Schmidt.
+    elr : bool, optional
+        If `True` (default), then extended local orthogonality is enforced
+        when obtaining singular vectors.
+    blocksize : int, optional
+        If computing `u` or `v` (left and right singular vectors,
+        respectively), `blocksize` controls how large a fraction of the
+        work is done via fast BLAS level 3 operations.  A larger blocksize
+        may lead to faster computation at the expense of greater memory
+        consumption.  `blocksize` must be ``>= 1``.  Default is `16`.
+    min_relgap : float, optional
+        The smallest relative gap allowed between any shift in IRL mode.
+        Default is `0.001`.  Accessed only if ``irl_mode=True``.
+    shifts : int, optional
+        Number of shifts per restart in IRL mode.  Default is determined
+        to satisfy ``k <= min(kmax-shifts, m, n)``.  Must be
+        >= 0, but choosing 0 might lead to performance degredation.
+        Accessed only if ``irl_mode=True``.
+    maxiter : int, optional
+        Maximum number of restarts in IRL mode.  Default is `1000`.
+        Accessed only if ``irl_mode=True``.
+
+    Returns
+    -------
+    u : ndarray
+        The top `k` left singular vectors, ``shape == (A.shape[0], 3)``,
+        returned only if ``compute_u=True``.
+    sigma : ndarray
+        The top `k` singular values, ``shape == (k,)``
+    vt : ndarray
+        The top `k` right singular vectors, ``shape == (3, A.shape[1])``,
+        returned only if ``compute_v=True``.
+    sigma_bound : ndarray
+        the error bounds on the singular values sigma, returned only if
+        ``full_output=True``.
+
+    Examples
+    --------
+    >>> from scipy.sparse.linalg import svdp
+    >>> A = np.random.random((10, 20))
+    >>> u, sigma, vt = svdp(A, 3)
```

Comment:
If it sounds good to you, I'll add an example that when `k=10`, `np.isclose(u @ np.eye(sigma) @ vt, A)` is `True`? And maybe for `k < 10`, show that they are roughly the same. (Maybe I'll construct the matrix to make a low-dimension approximation a good one.)

Reply:
That would be a good SVD demo in general, does that make more sense in the top `svds` function doctest?

Reply:
Done in gh-14251.